platforms/core-configuration/kotlin-language-extensions/build.gradle.kts

```diff
@@ -14,11 +14,13 @@
  * limitations under the License.
  */
 
-package org.gradle.configurationcache.serialization.beans
-
-import org.gradle.configurationcache.serialization.WriteContext
+plugins {
+    id("gradlebuild.distribution.implementation-kotlin")
+}
 
+description = "Useful extension methods for Kotlin"
 
-interface BeanStateWriter {
-    suspend fun WriteContext.writeStateOf(bean: Any)
+dependencies {
+    api(libs.futureKotlin("stdlib"))
+    implementation(projects.javaLanguageExtensions)
```

Comment:
Now that I see the parallel between this new module and the existing `java-language-extensions`, I have no issue with the new name. Actually, I think this module is a much better described as "stdlib-extensions" than the one for Java, as the latter actually introduces things that go way beyond providing general purpose extensions to the Java library. 

It seems we are consuming `java-language-extensions` just to get access to `Cast`. That is fine, more of a problem with that module's purpose than with this one (and that module's scope is already an improvement over recent past).